packages/gatsby/src/schema/__tests__/rebuild-sitepage-type.js

```diff
@@ -157,8 +161,16 @@ describe(`build and update schema for SitePage`, () => {
         keep: String!
         fields: SitePageFields
       }
+      type SitePageFields {
+        temp: String!
+      }
```

Comment:
This is needed because of new `deleteFieldsOfDerivedTypes` function? (As in it was kind of broken that test was working as is without haveing `SitePageFields` definition?)

Reply:
Yes, this test was not actually testing anything before 😄

Reply:
Heh, I just realized that it was working before because inference metadata does not ignore `CREATE_TYPES` (and adds type with `@dontInfer` to the list of ignored types. This way it was kinda working as expected.

But those definitions were ignored in all `schema.js` related calls. And now that we also have `deleteFieldsOfDerivedTypes` we actually need to fully rebuild the type. And unless we also define `SitePageFields` this `build` call will fail because there is no such type in the schema anymore.